doc/mvf_v4.rst

```diff
@@ -8,12 +8,441 @@ the data is split into chunks, each in its own file, which are loaded
+    forming filenames and telstate namespaces) and, where necessary, a sequence
+    number appended to ensure uniqueness.
+
+.. _katpoint: https://github.com/ska-sa/katpoint
+
+For each ``sdp.continuum_image`` stream, there is a sub-namespace per target
+(named with the normalised target name) with the following keys (keeping
+in mind that ``.`` is used to indicate whichever separator is in use by
+katsdptelstate for this database):
+
+``target0.clean_components`` (dict)
+    Image of the target field as a set of point sources. The ``target0``
+    sub-namespace is used to allow for possible alternative ways to run
+    the continuum imager in which a single execution would image
+    multiple fields, in which case there would be :samp:`target{N}`
+    sub-namespaces up to some *N*. This is not currently expected for
+    MeerKAT science observations.
+
+    The dictionary has two keys:
+
+    ``description`` (string)
+        `katpoint`_ description of the target field (specifically, the
+        phase centre).
+
+    ``components`` (list of string)
+        `katpoint_` target descriptions for the CLEAN components. The
+        names are arbitrary. This describes the **perceived** sky i.e., are
+        modulated by the primary beam.
+
+
+:samp:`m{NNN}.gains` (array)
+    TODO: is this the final form for the selfcal solutions?
+
+.. _linking-streams:
+
+.. _timestamps:
+
+Timestamps
+^^^^^^^^^^
+
+Timestamps are not stored explicitly. Instead, the first timestamp and
+the interval between dumps are stored, from which timestamps can be
+synthesised. The ith dump has a central timestamp (in the UNIX epoch) of
+:math:`\text{sync_time} + \text{first_timestamp} + i \times
+\text{int_time}`. The split of the initial timestamp into two parts is
+for technical reasons.
+
+There is also ``first_timestamp_adc``, which is the same as
+``first_timestamp`` but in units of the digitiser ADC counts. It is
+stored only for internal implementation reasons and should not be relied
+upon.
+
+.. _data:
+
+Data
+----
+
+Visibilities, flags and weights are subdivided into small *chunks*. The
+chunking model is based on `dask`_. Visibilities are treated as a 3D
+array, with axes for time, frequency and baseline. The data is divided
+into pieces along each axis. Each piece is stored in a separate file
+in the archive, in `.npy format`_. The metadata necessary to reconstruct
+the array is stored in the telescope state and documented in more detail
+later. It is possible that some chunks will be missing, because they
+were lost during the capture process. On load, katdal will replace such
+chunks with default values and set the ``data_lost`` flag for them.
+Weights and flags are similarly treated.
+
+.. _dask: http://docs.dask.org/en/latest/
+
+.. _.npy format: https://docs.scipy.org/doc/numpy-1.14.0/neps/npy-format.html
+
+Chunks are named :samp:`{type}/{AAAAA}_{BBBBB}_{CCCCC}.npy` where *type*
```

Comment:
Not guaranteed to be exactly five digits each, but expected to be with typical MeerKAT data...

I use `%05d` to map integers to strings, so they are at least five digits and padded with zeros.
The time index needs the pathological case of 27 hours of 1-second dumps to extend beyond five digits, we need a different correlator to get to 128k channels and the baseline chunks are typically not split, so I reckon it's a safe assumption.